luatest/treegen.lua

```diff
@@ -0,0 +1,210 @@
+--- Working tree generator.
+--
+-- Generates a tree of Lua files using provided templates and
+-- filenames.
+--
+-- @usage
+--
+-- local t = require('luatest')
+-- local treegen = require('test.treegen')
+--
+-- local g = t.group()
+--
+-- local SCRIPT_TEMPLATE = [[
+--     <...>
+-- ]]
+--
+-- g.before_all(function(g)
+--     treegen.init(g)
+--     treegen.add_template(g, '^.*$', SCRIPT_TEMPLATE)
+-- end)
+--
+-- g.after_all(function(g)
+--     treegen.clean(g)
+-- end)
+--
+-- g.foobar_test = function(g)
+--     local dir = treegen.prepare_directory(g,
+--         {'foo/bar.lua', 'main.lua'})
+--     <..test case..>
+-- end
+--
+-- @module luatest.treegen
+
+local hooks = require('luatest.hooks')
+
+local fio = require('fio')
+local fun = require('fun')
+local checks = require('checks')
+
+local log = require('luatest.log')
+
+local treegen = {}
+
+local function find_template(group, script)
+    for _, template_def in ipairs(group.templates) do
+        if script:match(template_def.pattern) then
+            return template_def.template
+        end
+    end
+    error(("treegen: can't find a template for script %q"):format(script))
+end
+
+--- Write provided script into the given directory.
+--
+-- @string directory Directory where the script will be created.
+-- @string filename File to write (possible nested path: /foo/bar/main.lua).
+-- @string script The body of the script to write.
+-- @return string
+function treegen.write_script(directory, filename, script)
```

Comment:
This function was initially written for writing/generating scripts, but now it is used for configuration files too. How about a more general name? For example, `write_file`?

Reply:
Yep, renamed it.